### ISSUE TITLE
Curious about performance of multiple optimizers vs single optimizer

### ISSUE THREAD

**Issue by seandickert** [mravanelli/SincNet#35]
Just curious if you ever tested using a single optimizer for all the networks, and if so, was the performance worse? Similarly, have you run any experiments varying the learning rate or other optimizer parameters per layer?

Thanks!

Sean

**Comment by mravanelli** [comment]
Hi Sean,
are you talking about the SincNet paper (https://arxiv.org/abs/1808.00158)
or my latest work (https://arxiv.org/abs/1812.00271)?
In general, if you would like to play with different optimization stuff you
can take a look into the SincNet implementation within the PyTorch-Kaldi
project (https://github.com/mravanelli/pytorch-kaldi). Pytorch-Kaldi can
define different optimizers for each neural netowork in your model.

Mirco

**Comment by seandickert** [comment]
I was referencing the earlier SincNet paper, and specifically, the code in this repo. The current setup uses 3 separate optimizers for the 3 separate nets (CNN, DNN1, DNN2). Currently, they all use the same hyper parameters. I was just curious if you ever looked at varying the parameters for the different nets. For example, learning rate of .001 for CNN but .01 for DNN1, etc.

**Comment by mravanelli** [comment]
Yes, we did it to allow more flexibility to the users that can possibly
explore different optimization strategies on different networks. In this
study, we didn't do it, but could be interesting to try..



<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>
